crates/libm-cdylib/src/macros.rs

```diff
@@ -0,0 +1,72 @@
+macro_rules! export {
+    (fn $id:ident ($($arg:ident : $arg_ty:ty),* ) -> $ret_ty:ty:
+       ($($test_arg:expr),*) -> $test_ret:expr;) => {
+        #[no_mangle]
+        pub extern "C" fn $id($($arg: $arg_ty),*) -> $ret_ty {
+            // This just forwards the call to the appropriate function of the
+            // libm crate.
+            #[cfg(not(link_test))] {
```

Comment:
Adding a pretty complicated macro to test what already has a pretty complicated of tests seems like overkill? Why does the cdylib itself need to be tested when it's all already tested elsewhere?

Reply:
This tests that the C ABI we provide is compatible with `<math.h>`, which is what allows this library to be used from C. 

We currently don't test it, and in fact, `libm` currently doesn't provide a compatible ABI, which is why some of the functions in the `lib.rs` are commented out. 

Simplest example is the `libm` functions returning tuples by value (e.g. `(i32, f32)`). The libm C ABI doesn't do that, and takes pointer arguments instead that are used as "out" parameters.